### ISSUE TITLE
LazyTree.zip/1 function clause error

### ISSUE THREAD

**Issue by aaronrenner** [whatyouhide/stream_data#84]
I'm getting intermittent FunctionClauseErrors in `LazyTree.zip/1` after upgrading to StreamData 0.4.1.

```
  1) property strip/1 removes script, noscript tags from dom
(StreamdataLazyTreeZipErrorTest)
     test/streamdata_lazy_tree_zip_error_test.exs:5
     ** (FunctionClauseError) no function clause matching in anonymous fn/1 in
StreamData.LazyTree.zip/1

     The following arguments were given to anonymous fn/1 in
StreamData.LazyTree.zip/1:

         # 1
         false

     code: check all document_node <- document_node(), max_runs: 15 do
     stacktrace:
       (stream_data) lib/stream_data/lazy_tree.ex:151: anonymous fn/1 in
StreamData.LazyTree.zip/1
       (elixir) lib/enum.ex:1255: Enum."-map/2-lists^map/1-0-"/2
       (stream_data) lib/stream_data/lazy_tree.ex:151: StreamData.LazyTree.zip/1
       (elixir) lib/stream.ex:516: anonymous fn/4 in Stream.map/2
       (elixir) lib/enum.ex:3161: Enumerable.List.reduce/3
       (elixir) lib/stream.ex:849: Stream.do_list_transform/10
       (elixir) lib/stream.ex:1433: Enumerable.Stream.do_each/4
       (elixir) lib/stream.ex:870: Stream.do_enum_transform/10
       (elixir) lib/stream.ex:1433: Enumerable.Stream.do_each/4
       (elixir) lib/stream.ex:870: Stream.do_enum_transform/10
       (elixir) lib/stream.ex:806: Stream.do_transform/8
       (elixir) lib/stream.ex:1433: Enumerable.Stream.do_each/4
       (elixir) lib/stream.ex:870: Stream.do_enum_transform/10
       (elixir) lib/stream.ex:806: Stream.do_transform/8
       (elixir) lib/stream.ex:1433: Enumerable.Stream.do_each/4
       (elixir) lib/stream.ex:870: Stream.do_enum_transform/10
       (stream_data) lib/stream_data.ex:2045: StreamData.shrink_failure/6
       (stream_data) lib/stream_data.ex:2017: StreamData.check_all/7
       test/streamdata_lazy_tree_zip_error_test.exs:6: (test)
```

Here's a project that you should be able to reproduce the issue with: https://github.com/aaronrenner/stream-data-lazy-tree-zip-error. I've seen this issue with OTP 20.0 and 20.2 and Elixir 1.5.1 and Elixir 1.5.3. I haven't tested it on any other versions.

**Comment by whatyouhide** [comment]
Hey @aaronrenner, the project is very complex for trying to find out a cause for this problem. Any chance you can find a smaller case that reproduces the bug?

**Comment by aaronrenner** [comment]
I tried simplifying it, but it wouldn't produce the bug consistently. Were you able to run the code and get this error? I can spend some time in the evening to see how much I can remove before I have trouble recreating the error.

**Comment by whatyouhide** [comment]
Hey @aaronrenner, I just closed #90 which I believe was caused by the same thing causing the issue reported here. I have cloned the repo you pointed out and tried `mix test` many times and never encountered this same error. I'm gonna close this and consider it fixed, please feel free to reopen if you still stumble upon this bug :) Thanks for the report! 💟

**Comment by aaronrenner** [comment]
Looks like this issue is fixed in 0.4.2. Thanks so much! ❤️
